cmd/eval.go

```diff
@@ -83,10 +85,17 @@ Example: combo eval -r REPLACE_ME=1,2,3 path/to/file
 				return fmt.Errorf("failed to validate file specified: %w", err)
 			}
 
-			combinations := combination.NewStream(
-				combination.WithArgs(formatReplacements(replacements)),
-				combination.WithSolveAhead(),
-			)
+			var combinations combination.Stream
+			if useSolvedAhead {
+				combinations = combination.NewStream(
+					combination.WithArgs(formatReplacements(replacements)),
+					combination.WithSolveAhead(),
+				)
+			} else {
+				combinations = combination.NewStream(
+					combination.WithArgs(formatReplacements(replacements)),
+				)
+			}
```

Comment:
I think this might be a candidate for a slight refactor - we could update that `combination.WithSolveAhead()` StreamOption to take a bool, and that method is responsible for setting the value of the `cs.solveAhead = solveAhead`. @tylerslaton any thoughts? We'd be able to remove the need to conditionally check the `useSolveAhead` variable here, and remove the nesting that reduces readability.

Reply:
Another alternative could be instead of using `WithSolveAhead()` we create getter and setter methods such as `setSolveAhead()` and `getSolveAhead()`, so there is no confusion on why `WithSolveAhead()` that is currently used to set solve ahead as `true` takes a variable.

Reply:
Yeah that's one possibility - I'm generally a big fan of this functional options pattern as it helps maintain API flexibility, but the discoverability of the various options you can use isn't always ideal, especially if those individual options aren't prefixed with `With*(...)`. I'd like to avoid any getter/setters in the API as that feels like an anti-pattern for Go these days.

Anyways, not a huge deal as this project is still in the early dev stages, so we can always re-evaluate our API over time before committing to anything long term.

Reply:
For sure, I know I originally had made that change to simplify the code, but Tyler and I ended up deciding it could end up being confusing.

Reply:
Yeah, Jon had originally made a change where `WithSolveAhead()` took a bool and this got cleaned up. I was originally opposed to it, but I think that this does become cleaner to a degree that makes sense now that we're talking about it here.

Reply:
Alternatively, we could go with an "optional parameter" approach:

https://stackoverflow.com/questions/19612449/default-value-in-gos-method

Reply:
I like the optional approach. 
Here's how it would work inline
```
func WithSolveAhead(boolVar ...bool) StreamOption {
	boolVarValue := false
	if len(boolVar) > 0 {
		boolVarValue = boolVar[0]
	}
	return func(cs *stream) {
		cs.solveAhead = boolVarValue
	}
}
```

Reply:
I like that, @timflannagan thoughts?

Reply:
I think that implementation is introducing extra complexity, but let's just forward on whatever we're comfortable with now. I wouldn't block a PR on this.

Reply:
Let's just go with the required boolean (like what Jon had before) and call it a day on this thread then.

Reply:
Got it, I committed with optional, but I will re commit with the required.